tests/workflows/test_snowflake.py

```diff
@@ -0,0 +1,163 @@
+                start_lat number,
+                start_lng number,
+                end_lat number,
+                end_lng number,
+                is_member boolean not null
+            )"""
+    )
+    yield name
+
+
+@pytest.mark.client("snowflake")
+def test_exploratory_analysis(client, connection_kwargs, table):
+    """Upload NYC bike dataset to Snowflake. Then read
+    and explore the data."""
+
+    csv_path = "nyc-bike-data"
+
+    def unzip(filename: str):
+        """Unzip csv file to csv_path"""
+        if not os.path.exists(f"{csv_path}/{filename}"):
+            url = f"https://s3.amazonaws.com/tripdata/{filename}.zip"
+            zip_path, _ = urllib.request.urlretrieve(url)
+            with zipfile.ZipFile(zip_path, "r") as f:
+                f.extractall(csv_path)
+        return filename
+
+    filenames = [
+        f"{ts.year}{ts.month:02}-citibike-tripdata.csv"
+        for ts in pd.date_range(start="2022-01-01", end="2023-03-01", freq="MS")
+    ]
+
+    # download and unzip files
+    futures = client.map(unzip, filenames)
+    wait(futures)
+
+    # preprocess data
+    def safe_int(x):
+        """Some station IDs are not correct integers"""
+        try:
+            return int(float(x))
+        except Exception:
+            # if station ID is not an int, return -1
+            return -1
+
+    ddf = dd.read_csv(
+        f"{csv_path}/*.csv",
+        converters={"start_station_id": safe_int, "end_station_id": safe_int},
+    )
+
+    # filter out incorrect station IDs
+    ddf = ddf[(ddf.start_station_id != -1) & (ddf.end_station_id != -1)].reset_index(
+        drop=True
+    )
+
+    # create boolean is_member and drop member_casual
+    ddf["is_member"] = ddf.member_casual.apply(
+        lambda x: x == "member", meta=("is_member", bool)
+    )
+    ddf = ddf.drop(columns="member_casual")
+
+    # repartition to ensure even chunks
+    ddf = ddf.repartition(partition_size="100Mb")
+
+    input_npartitions = ddf.npartitions
+
+    wait(ddf)
+
+    # save data to Snowflake
+    to_snowflake(ddf, name=table, connection_kwargs=connection_kwargs)
+
+    # read the data back
+    ddf_out = read_snowflake(
+        f"SELECT * FROM {table}",
+        connection_kwargs=connection_kwargs,
+        npartitions=input_npartitions,
+    ).rename(
+        columns=str.lower
+    )  # snowflake has all uppercase
+
+    ddf_out["is_roundtrip"] = ddf_out["start_station_id"] == ddf_out["end_station_id"]
+
+    # compute counts of roundtrips vs non-roundtrips
+    roundtrips = ddf_out.is_roundtrip.value_counts().compute()  # noqa: F841
+
+    """
+    # maybe plot roundtrips vs one-way
+    import seaborn as sns
+    sns.barplot(roundtrips, x="is_roundtrip", y="count")
+    """
+
+    # explore trip duration
+    ddf_out["trip_duration"] = ddf_out["ended_at"] - ddf_out["started_at"]
+    ddf_out["trip_minutes"] = ddf_out["trip_duration"].apply(
+        lambda x: x.total_seconds() // 60, meta=("trip_duration", int)
```

Comment:
Using the apply here is probably going to be very slow.  I recommend using  `.dt.total_seconds` instead.  In general any time pandas uses apply it runs at Python rather than Cython speeds, and so is very slow.